src/Passes/Source/Passes/QubitAllocationAnalysis/QubitAllocationAnalysis.cpp

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Passes/QubitAllocationAnalysis/QubitAllocationAnalysis.hpp"
+
+#include "Llvm/Llvm.hpp"
+
+#include <fstream>
+#include <iostream>
+#include <unordered_set>
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    QubitAllocationAnalysisAnalytics::Result QubitAllocationAnalysisAnalytics::run(
+        llvm::Function& function,
+        llvm::FunctionAnalysisManager& /*unused*/)
+    {
+        for (auto& basic_block : function)
+        {
+            for (auto& instr : basic_block)
+            {
+                auto call_instr = llvm::dyn_cast<llvm::CallBase>(&instr);
+                if (call_instr == nullptr)
+                {
+                    continue;
+                }
+                auto target_function = call_instr->getCalledFunction();
+                auto name            = target_function->getName();
+                //      llvm::errs() << "Testing " << name << " : " << *call_instr << "\n";
+
+                if (name == "__quantum__rt__qubit_allocate")
+                {
+                    return {true};
+                }
+
+                if (name == "__quantum__rt__qubit_allocate_array")
+                {
+                    return {true};
+                }
+
+                if (name == "__quantum__qis__m__body")
```

Comment:
Yes, I agree that that is a better name. It started out just scanning for qubit allocations until I realised that I also needed to capture result allocations.

Reply:
I've renamed it to `QirAllocationAnalysis` as this does not suggest which types are meant to be captured, but just that it relates to the Qir. I.e. it is not just qubits and it is not all opaque types (although it might be).